python/ray/experimental/dask/scheduler.py

```diff
@@ -73,23 +77,48 @@ def ray_dask_get(dsk, keys, **kwargs):
                 atexit.register(pool.close)
                 pools[thread][num_workers] = pool
 
-    # NOTE: We hijack Dask's `get_async` function, injecting a different task
-    # executor.
-    object_refs = get_async(
-        _apply_async_wrapper(pool.apply_async, _rayify_task_wrapper),
-        len(pool._pool),
-        dsk,
-        keys,
-        get_id=_thread_get_id,
-        pack_exception=pack_exception,
-        **kwargs,
-    )
-    # NOTE: We explicitly delete the Dask graph here so object references
-    # are garbage-collected before this function returns, i.e. before all Ray
-    # tasks are done. Otherwise, no intermediate objects will be cleaned up
-    # until all Ray tasks are done.
-    del dsk
-    result = ray_get_unpack(object_refs)
+    ray_callbacks = kwargs.pop("ray_callbacks", None)
+
+    with local_ray_callbacks(ray_callbacks) as ray_callbacks:
+        # Unpack the Ray-specific callbacks.
+        (
```

Comment:
It might be nice to make this a namedtuple of callbacks, just to reduce verbosity and guard against future changes.